### PR TITLE
Add namespace formatting rules

### PR DESCRIPTION
**Pull request by ryzngard**
## Summary

Adds new namespace formatting rules to the documentation with examples on how they are used

### CODE REVIEW

File: docs/fundamentals/code-analysis/style-rules/formatting-rules.md [hunk 1]
```diff
@@ -110,6 +112,43 @@ using System.Threading.Tasks;
 using Octokit;
 ```
 
+### Dotnet namespace options
+
```
[the remark this answers is not in the document]
Comment (ryzngard) [reply docs/fundamentals/code-analysis/style-rules/formatting-rules.md:116]: Good catch. I thought I copied it, but I didn't

**Review by gewarren — APPROVED** [review]
Thanks @ryzngard

File: docs/fundamentals/code-analysis/style-rules/formatting-rules.md [hunk 2]
```diff
@@ -110,6 +112,53 @@ using System.Threading.Tasks;
...
+These formatting rules concern the declaration of `Namespace` blocks in both C# and Visual Basic.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/formatting-rules.md:117]:
```suggestion
These formatting rules concern the declaration of namespaces in both C# and Visual Basic.
```

File: docs/fundamentals/code-analysis/style-rules/formatting-rules.md [hunk 3]
```diff
@@ -110,6 +112,53 @@ using System.Threading.Tasks;
...
+[*.{cs,vb}]
+dotnet_style_namespace_match_folder = true
+```
+
+#### dotnet\_style\_namespace\_match\_folder
+
+|Property|Value|
+|-|-|
+| **Option name** | dotnet_style_namespace_match_folder |
+| **Applicable languages** | C# |
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/formatting-rules.md:132]:
```suggestion
| **Applicable languages** | C# and Visual Basic |
```

**Comment by gewarren** [comment]
Close/reopen for build.

**Pull request merged**